packages/core/content-manager/admin/src/utils/validation.ts

```diff
@@ -98,11 +98,28 @@ const createYupSchema = (
               return {
                 ...acc,
                 [name]: transformSchema(
-                  yup.array().of(
-                    yup.object().shape({
-                      id: yup.string().required(),
-                    })
-                  )
+                  yup.lazy((value) => {
+                    if (!value) {
+                      return yup.mixed().nullable(true);
+                    } else if (Array.isArray(value)) {
```

Comment:
I don't think this will ever be the case in the admin? 🤔  Interested to hear why you added it

Reply:
Wasn't an array the only case here before this PR?

The main thing I am trying to add is

```
          } else if (typeof value === 'object') {
                      // TODO assert for specific keys such as connect, disconnect, count?
                      return yup.object();
                    } else {

```

Because otherwise the bulk locale publish modal displays errors for relation
fields (caused by the useDocument validate) when the value in the document is something like { count: 0 } 

e.g. the e2e tests were failing here
https://github.com/strapi/strapi/blob/e93e2f954ae682623665528511b8af3eb0784cb8/tests/e2e/tests/i18n/editview.spec.ts#L311

Because of the empty author relation